### PR TITLE
ci: add OSV validator for `json` files

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add `OSV` validator + `vulns.md` with small info about `vulns` dir

test runs:
- invalid JSON - https://github.com/DmitriyLewen/vuln-list-aqua/actions/runs/12425860570/job/34693267819
- invalid purl - https://github.com/DmitriyLewen/vuln-list-aqua/actions/runs/12425838443/job/34693208165

### CODE REVIEW

File: vulns/vulns.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# vulns
+The `vulns` directory contains advisories in [OSV][osv] format added by Aquasecurity.
```
Comment (knqyf263) [inline vulns/vulns.md]: nit
```suggestion
The `vulns` directory contains advisories in [OSV][osv] format added by Aqua Security.
```

File: .github/workflows/validate.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # For git diff
+
+      - name: Download the JSON schema
+        run: wget https://raw.githubusercontent.com/ossf/osv-schema/v1.6.7/validation/schema.json
+
+      - name: Install check-jsonschema
+        run: pip install check-jsonschema
```
Comment (knqyf263) [inline .github/workflows/validate.yaml]: We might be able to leverage this tool (I recently found).
https://github.com/ossf/osv-schema/tree/main/tools/osv-linter

Comment (DmitriyLewen) [reply .github/workflows/validate.yaml]: updated in https://github.com/aquasecurity/vuln-list-aqua/pull/2/commits/ded9b95ab852f0a0e98c5473be1969177673250c

File: .github/workflows/validate.yaml [hunk 3]
```diff
@@ -12,18 +13,30 @@ jobs:
...
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 'stable'
 
-      - name: Install check-jsonschema
-        run: pip install check-jsonschema
+      - name: Install osv-linter
+        run: |
+          git clone https://github.com/ossf/osv-schema.git
```
Comment (knqyf263) [inline .github/workflows/validate.yaml]: Should pin the version

```suggestion
          git clone https://github.com/ossf/osv-schema.git -b v1.6.7
```

Comment (DmitriyLewen) [reply .github/workflows/validate.yaml]: make sense 👍 
pinned in https://github.com/aquasecurity/vuln-list-aqua/pull/2/commits/af4136e8734288ab66aa180f069ff9c29255fa79

File: .github/workflows/validate.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          cd ../../..
+          rm -rf osv-schema
+
+      - name: Validate JSON
+        run: |
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | grep .json) && : // Ignore errors
+          if [ -n "$CHANGED_FILES" ]; then
+            echo "Validating JSON files: $CHANGED_FILES"
+            ## osv-linter doesn't return error code for invalid JSON. So we should handle this case.
+            output=$(./osv-linter record lint vulns/AQUA-2024-0001.json 2>&1 || true)
```
Comment (knqyf263) [inline .github/workflows/validate.yaml]: Need to use `$CHANGED_FILES` instead of AQUA-2024-0001.json.

Comment (DmitriyLewen) [reply .github/workflows/validate.yaml]: Thanks, I forgot to change that after testing on my local PC.

Fixed in https://github.com/aquasecurity/vuln-list-aqua/pull/2/commits/81d7fddebfb071bcff03cd8b81d914ebac02762d

Comment (knqyf263) [reply .github/workflows/validate.yaml]: Can `osv-linter` take several files as its argument?

Comment (DmitriyLewen) [reply .github/workflows/validate.yaml]: `osv-linter` works correctly with multiple files - https://github.com/DmitriyLewen/vuln-list-aqua/actions/runs/12426310367/job/34694377977

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**